helper/resource/error.go

```diff
@@ -0,0 +1,21 @@
+package resource
+
+type NotFoundError struct {
+	LastError    error
+	LastRequest  interface{}
+	LastResponse interface{}
+	Message      string
+	Retries      int
+}
+
+func (e *NotFoundError) Error() string {
+	if e.Message != "" {
+		return e.Message
+	}
+
+	return "couldn't find resource"
+}
+
+func NewNotFoundError(err string) *NotFoundError {
```

Comment:
For this use-case, I agree that the latter is clearer. For `RetryError`, given a choice between
```go
return resource.NonRetryableError(err)
```
and
```go
if err != nil {
  return &RetryError{Err: err, Retryable: false}
}
```
I'd say the first solution is much more readable. Maybe the helper is useful for the common case, and for more involved usage then instantiating manually is better?